pkg/onchain/storage/storage.go

```diff
@@ -30,22 +30,19 @@ func readZipFile(z *zip.File) ([]byte, error) {
 func Get(client *node.Client, address string, key string) (map[string][]byte, error) {
 	chunkNumberKey := "total_chunks"
 
-	keyNumber, err := node.DatastoreEntry(client, address, chunkNumberKey)
+	keyNumber, err := node.DatastoreEntry(client, address, []byte(chunkNumberKey))
 	if err != nil {
 		return nil, fmt.Errorf("reading datastore entry '%s' at '%s': %w", address, chunkNumberKey, err)
 	}
 
-	chunkNumber, err := strconv.Atoi(string(keyNumber.CandidateValue))
-	if err != nil {
-		return nil, fmt.Errorf("error converting String to integer")
-	}
+	chunkNumber := binary.LittleEndian.Uint64(keyNumber.CandidateValue)
 
 	entries := []node.DatastoreEntriesKeysAsString{}
 
-	for i := 0; i < chunkNumber; i++ {
+	for i := uint64(0); i < chunkNumber; i++ {
```

Comment:
because chunkNumber is uint64 and when you try to do i<chunkNumber linter is shouting

Reply:
is it realistic to use a U64 to store the number of chunks ?
In any case, cast chunkNumber or even better change its type because here you need to cast i twice (line 42 and line 45)